### ISSUE TITLE
🐛 Bug Report: SonarQube plugin fails on config check

### ISSUE THREAD

**Issue by Ritesh007** [backstage/backstage#18240]
### 📜 Description

When using the following config in the app config file the app runs as expected but when we run the 'yarn backstage-cli config:check --config app-config.yaml --lax --strict' it errors out says the config is not right.
```
sonarQube:
  baseUrl: https://xyz.com
  apiKey: <SONAR_TOKEN>
```

Error:
Error: Configuration does not match schema
  Config must NOT have additional properties { additionalProperty=apiKey } at /sonarQube

### 👍 Expected behavior

It shouldn't error out on the following config when the config check runs

```
sonarQube:
  baseUrl: https://xyz.com
  apiKey: <SONAR_TOKEN>
```

### 👎 Actual Behavior with Screenshots

It actually looks at the config of the sonar plugin frontend: https://github.com/backstage/backstage/blob/master/plugins/sonarqube/config.d.ts and not the backend config: https://github.com/backstage/backstage/blob/master/plugins/sonarqube-backend/config.d.ts.

### 👟 Reproduction steps

1. Install Sonar frontend and backend plugins.
2. Add the config to the app config.
3. Run 'yarn backstage-cli config:check --config app-config.yaml --lax --strict'

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

Yes I am willing to submit a PR!

**Comment by adamdmharvey** [comment]
@Ritesh007 and I chatted briefly on this. I wonder if the fix is to REMOVE the configuration schema entirely from the frontend plugin, which uses it to generate APIs to SonarQube, and leave it only in the backend plugin with the `@visibility frontend` directive?  i feel like the `--strict` check may just be picking it up from the frontend w/o doesn't use the `apiKey`, hence the error.

thoughts?

**Comment by Rugvip** [comment]
The issue is that the SonarQube backend plugin config isn't properly included in the package. That was fixed in #17850 and will be released into main-line next week. Closing as fixed.
